# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

### 1. What test is appropriate for this problem? Does CLT apply?
Two-sample z test is appropriate for this problem. The mean receiving employer call for white-sounding and black-sounding names.  CLT still applies even when the underlying distribution is not normal provided that sufficiently large sample > 30.  Obviously CLT applies on this data set.

### 2. What are the null and alternate hypotheses?
The hypothesis is constructed as below:
H0: Mean of receiving employer calls for white-sounding names = black-sounding names

H1: Mean of receiving employer calls for white-sounding names != black-sounding names

### 3. Compute margin of error, confidence interval, and p-value.

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [5]:
black_names = data[data.race=='b'].call
white_names = data[data.race=='w'].call

In [6]:
print('Mean receiving employer call for black-sounding names:', black_names.mean())
print('Mean receiving employer call for white-sounding names:', white_names.mean())
print('Mean difference between black-sounding and white-sounding names:', black_names.mean() - white_names.mean())

Mean receiving employer call for black-sounding names: 0.0644763857126236
Mean receiving employer call for white-sounding names: 0.09650924056768417
Mean difference between black-sounding and white-sounding names: -0.03203285485506058


Setting at 95% of confidence level for two tails, z-score = 1.96 according to norm.ppf(0.975)

Margin of error = critical value (z score) x standard error of the two sample z test

In [7]:
margin_of_error = abs(stats.norm.ppf((1-0.95)/2)) * np.sqrt(black_names.var()/len(black_names) + white_names.var()/len(white_names))
print('Margin of error at confidence level 95%:', margin_of_error)

Margin of error at confidence level 95%: 0.0152581370037


In [8]:
z_score = (black_names.mean() - white_names.mean() - 0) / np.sqrt(black_names.var()/len(black_names) + white_names.var()/len(white_names))
print("p-value:", norm.cdf(z_score)*2)

p-value: 3.87618807011e-05


### 4. Write a story describing the statistical significance in the context or the original problem.
From the above means of receiving employer calls for white-sounding and black-sounding names, mean of black-sounding names is 0.032 less than white-sound names and therefore black-sounding names are 3.2% less likely to be called by an employer than the white-sounding names.  Using the two sample z statistic, p-value is very small and we reject the null hypothesis to conclude that with more than 95% confidence, black-sounding names have a lower chance of getting a callback than white-sounding names.


### 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?
According to above statistical inference, race / name is a factor but not be able to conclude as the most important factor in callback success because we need to consider other features e.g. age, years of experience, gender, education level, expected salary etc. which may provide more significant indication on the callback success.  Therefore, more features should be included in the analysis in order to conclude the most important factor in callback success.
